From d9680d29f71d1608af6e355a0f27d7244c69efb4 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 20 May 2020 10:48:41 +0300
Subject: [PATCH] [SW-9973] spi: execute spi_prot() only once

The protection does not need to be ensured each time the spi_copy()
function is called. So do it only once.
Condition the spi_port() call for only one time (per boot) execution.
Since the print_wp_status variable now also controls the call to
spi_prot() function, rename it to better reflect its purpose.

Change-Id: Iabf90ea2600ab60a060dec6fe28ce80a3157248f
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/spi.c b/common/spi.c
index 8e579cd..3165103 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -234,15 +234,15 @@
 void spi_copy(void *dst, unsigned int offset, unsigned int size, char *name)
 {
 	unsigned char sr;
-	static bool print_wp_status = true;
+	static bool prot_flash = true;
 
 	spi_init();
 	spi_read(dst, offset, size);
-	sr = spi_prot();
 	preboot_puts_raw(name);
 	preboot_puts(" copying from flash is completed\n");
-	if (print_wp_status) {
-		print_wp_status = false;
+	if (prot_flash) {
+		prot_flash = false;
+		sr = spi_prot();
 		if ((sr & PROT_MASK) == PROT_MASK)
 			preboot_puts("Flash may be protected, depend on WP# signal\n");
 		else
